File: Main/ReferenceDLLs/Pack1/Facebook Visual Kit VB/FVK Source Code/FVK/FVK/LikeButton.ascx.vb
' WARNING: 
' The source code of this class can be used and modified to your particular needs. 
' Sharing the class with unregistered users or using it with unregistered websites 
' or Facebook applications is not allowed and violates the rights of intellectual 
' property. 

Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Text

Namespace FVK
    ''' <summary>
    ''' Like button allows users to share some content from the website with their friends. Pressing on 
    ''' the Like button will result in publishing of a story on user's profile, with link to the website.
    ''' </summary>
    Partial Public Class LikeButton
        Inherits System.Web.UI.UserControl
        ''' <summary>
        ''' URL of the like button. If it's not set, current page URL is used.
        ''' </summary>
        Public WriteOnly Property Url() As String
            Set(ByVal value As String)
                m_url = value
            End Set
        End Property



        ''' <summary>
        ''' Layout type. Available values are 'standard', 'button_count', 'box_count'
        ''' </summary>
        Public WriteOnly Property Layout() As String
            Set(ByVal value As String)
                If value = "button_count" Then
                    m_layout = value
                ElseIf value = "box_count" Then
                    m_layout = value
                ElseIf value <> "standard" Then
                    Throw New Exception("LikeButton Error:  Unsupported layout: " & value)
                End If
            End Set
        End Property

        ''' <summary>
        ''' If set to true, send button will be shown next to the like button
        ''' </summary>
        Public WriteOnly Property Send() As Boolean
            Set(ByVal value As Boolean)
                m_send = value
            End Set
        End Property


        ''' <summary>
        ''' Set to true to show profile picture bellow the like button. Default value is true.
        ''' </summary>
        Public WriteOnly Property ShowFaces() As Boolean
            Set(ByVal value As Boolean)
                m_showFaces = value
            End Set
        End Property


        ''' <summary>
        ''' Width of the like button in pixels. Default value is 450.
        ''' </summary>
        Public WriteOnly Property Width() As Integer
            Set(ByVal value As Integer)
                m_width = value
            End Set
        End Property

        ''' <summary>
        ''' Text of the like button. Available values are 'like' and 'recommend'. 
        ''' Defined text is translated depending on user language settings.
        ''' </summary>
        Public WriteOnly Property Action() As String
            Set(ByVal value As String)
                If value = "recommend" Then
                    m_action = value
                ElseIf value <> "like" Then
                    Throw New Exception("LikeButton Error:  Unsupported action: " & value)
                End If
            End Set
        End Property

        ''' <summary>
        ''' Font inside the like button. Available values are 'arial', 'lucida grande',
        ''' 'segoe ui', 'tahoma', 'trebuchet ms' and 'verdana'.
        ''' </summary>
        Public WriteOnly Property Font() As String
            Set(ByVal value As String)
                If value IsNot Nothing Then
                    Select Case m_font
                        Case "arial", "lucida grande", "segoe ui", "tahoma", "trebuchet ms", "verdana"
                            m_font = value
                            Exit Select
                        Case Else
                            Throw New Exception("Like Button Error:  Unsupported font: " & value)
                    End Select
                End If
            End Set
        End Property

        ''' <summary>
        ''' Color scheme. Available values are 'light' and 'dark'. Deault value is 'light'.
        ''' </summary>
        Public WriteOnly Property ColorScheme() As String
            Set(ByVal value As String)
                If value = "dark" Then
                    m_colorScheme = value
                ElseIf value <> "light" Then
                    Throw New Exception("LikeButton Error:  Unsupported Color Scheme: " & value)
                End If
            End Set
        End Property

        ''' <summary>
        ''' The reference string for tracking referals.
        ''' </summary>
        Public WriteOnly Property Reference() As String
            Set(ByVal value As String)
                m_reference = value
            End Set
        End Property

        ''' <summary>
        ''' Page Load
        ''' </summary>
        ''' <param name="sender"></param>
        ''' <param name="e"></param>
        Protected Sub Page_Prerender(ByVal sender As Object, ByVal e As EventArgs) Handles Me.PreRender
            Refresh()
        End Sub

        Private Sub Refresh()
            Dim html As New StringBuilder()
            html.Append("<fb:like ")
            If m_url IsNot Nothing Then
                html.Append("href='" & m_url & "' ")
            End If
            If m_layout IsNot Nothing Then
                html.Append("layout='" & m_layout & "' ")
            End If
            If m_send Then
                html.Append("send='true' ")
            End If
            If m_showFaces = False Then
                html.Append("show_faces='false' ")
            End If
            If m_width <> 450 Then
                html.Append("width='" & m_width.ToString() & "' ")
            End If
            If m_action IsNot Nothing Then
                html.Append("action='" & m_action & "' ")
            End If
            If m_font IsNot Nothing Then
                html.Append("font='" & m_font & "' ")
            End If
            If m_colorScheme IsNot Nothing Then
                html.Append("colorscheme='" & m_colorScheme & "' ")
            End If
            If m_reference IsNot Nothing Then
                html.Append("ref='" & m_reference & "' ")
            End If
            html.Append("></fb:like>" & vbLf)
            likeSpan.InnerHtml = html.ToString()
        End Sub

        ' Private members
        Private m_url As String = Nothing
        Private m_layout As String = Nothing
        Private m_send As Boolean = False
        Private m_showFaces As Boolean = True
        Private m_width As Integer = 450
        Private m_action As String = Nothing
        Private m_font As String = Nothing
        Private m_colorScheme As String = Nothing
        Private m_reference As String = Nothing



        ' Obsolete members/methods

        <Obsolete("You shouldn't use this property anymore. Use ColorScheme instead")> _
        Public WriteOnly Property SetColorScheme() As String
            Set(ByVal value As String)
                If value = "dark" Then
                    m_colorScheme = value
                ElseIf value <> "light" Then
                    Throw New Exception("LikeButton Error:  Unsupported Color Scheme: " & value)
                End If
            End Set
        End Property
    End Class
End Namespace

File: Main/ReferenceDLLs/Pack1/Facebook Connect Visual Kit CS/FVK Source Code/FVK/FVK/LikeButton.ascx.cs
// WARNING: 
// The source code of this class can be used and modified to your particular needs. 
// Sharing the class with unregistered users or using it with unregistered websites 
// or Facebook applications is not allowed and violates the rights of intellectual 
// property. 

using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Text;

namespace FVK
{
    /// <summary>
    /// Like button allows users to share some content from the website with their friends. Pressing on 
    /// the Like button will result in publishing of a story on user's profile, with link to the website.
    /// </summary>
    public partial class LikeButton : System.Web.UI.UserControl
    {
        /// <summary>
        /// URL of the like button. If it's not set, current page URL is used.
        /// </summary>
        public string Url { set {url = value;} }

        

        /// <summary>
        /// Layout type. Available values are 'standard', 'button_count', 'box_count'
        /// </summary>
        public string Layout
        { 
            set {
                if (value == "button_count")
                {
                    layout = value;
                }
                else if (value == "box_count")
                {
                    layout = value;
                }
                else if (value != "standard")
                {
                    throw new Exception("LikeButton Error:  Unsupported layout: " + value);
                }
            }
        }

        /// <summary>
        /// If set to true, send button will be shown next to the like button
        /// </summary>
        public bool Send { set { send = value; } }
    

        /// <summary>
        /// Set to true to show profile picture bellow the like button. Default value is true.
        /// </summary>
        public bool ShowFaces { set { showFaces = value; } } 
        

        /// <summary>
        /// Width of the like button in pixels. Default value is 450.
        /// </summary>
        public int Width { set { width = value; } }
        
        /// <summary>
        /// Text of the like button. Available values are 'like' and 'recommend'. 
	    /// Defined text is translated depending on user language settings.
        /// </summary>
        public string Action
        {
            set 
            {
                if (value == "recommend")
                {
                    action = value;
                }
                else if (value != "like")
                {
                    throw new Exception("LikeButton Error:  Unsupported action: " + value);
                }
            }
        }

        /// <summary>
        /// Font inside the like button. Available values are 'arial', 'lucida grande',
        /// 'segoe ui', 'tahoma', 'trebuchet ms' and 'verdana'.
        /// </summary>
        public string Font
        {
            set 
            {
                if (value != null)
                {
                    switch (font)
                    {
                        case "arial":
                        case "lucida grande":
                        case "segoe ui":
                        case "tahoma":
                        case "trebuchet ms":
                        case "verdana":
                            font = value;
                            break;
                        default:
                            throw new Exception("Like Button Error:  Unsupported font: " + value);
                    }
                }
            }
        }

        /// <summary>
        /// Color scheme. Available values are 'light' and 'dark'. Deault value is 'light'.
        /// </summary>
        public string ColorScheme
        {
            set 
            {
                if (value == "dark")
                {
                    colorScheme = value;
                }
                else if (value != "light")
                {
                    throw new Exception("LikeButton Error:  Unsupported Color Scheme: " + value);
                }
            }
        }

        /// <summary>
        /// The reference string for tracking referals.
        /// </summary>
        public string Reference { set { reference = value; } }

        /// <summary>
        /// Page Load
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        protected void Page_Prerender(object sender, EventArgs e)
        {
            Refresh();
        }

        private void Refresh()
        {
            StringBuilder html = new StringBuilder();
            html.Append("<fb:like ");
            if (url != null)
            {
                html.Append("href='" + url + "' ");
            }
            if (layout != null)
            {
                html.Append("layout='" + layout + "' ");
            }
            if (send)
            {
                html.Append("send='true' ");
            }
            if (showFaces == false)
            {
                html.Append("show_faces='false' ");
            }
            if (width != 450)
            {
                html.Append("width='" + width.ToString() + "' ");
            }
            if (action != null)
            {
                html.Append("action='" + action + "' ");
            }
            if (font != null)
            {
                html.Append("font='" + font + "' ");
            }
            if (colorScheme != null)
            {
                html.Append("colorscheme='" + colorScheme + "' ");
            }
            if (reference != null)
            {
                html.Append("ref='" + reference + "' ");
            }
            html.Append("></fb:like>\n");
            likeSpan.InnerHtml = html.ToString();
        }

        // Private members
        private string url = null;
        private string layout = null;
        private bool send = false;
        private bool showFaces = true;
        private int width = 450;
        private string action = null;
        private string font = null;
        private string colorScheme = null;
        private string reference = null;



        // Obsolete members/methods

        [Obsolete("You shouldn't use this property anymore. Use ColorScheme instead")]
        public string SetColorScheme
        {
            set
            {
                if (value == "dark")
                {
                    colorScheme = value;
                }
                else if (value != "light")
                {
                    throw new Exception("LikeButton Error:  Unsupported Color Scheme: " + value);
                }
            }
        }
    }
}